tools/checkcomments/checkcomments.go

```diff
@@ -92,38 +94,31 @@ func run(pass *analysis.Pass) (any, error) {
 
 				match := todoRE.FindStringSubmatch(line)
 
-				if len(match) != 4 {
+				if len(match) != 2 {
 					pass.Reportf(c.Pos(), "invalid TODO: incorrect format")
 					continue
 				}
 
 				url := match[1]
-				repo := match[2]
-				num, err := strconv.Atoi(match[3])
-				if err != nil {
-					log.Panic(err)
-				}
-
-				if num <= 0 {
-					pass.Reportf(c.Pos(), "invalid TODO: incorrect issue number")
-					continue
-				}
 
-				if client == nil {
-					continue
-				}
+				status, err := client.IssueStatus(context.TODO(), url)
 
-				status, err := client.IssueStatus(context.TODO(), url, repo, num)
-				if err != nil {
+				switch {
+				case err == nil:
+					// nothing
+				case errors.Is(err, github.ErrIncorrectURL),
+					errors.Is(err, github.ErrIncorrectIssueNumber):
+					log.Print(err.Error())
+				default:
 					log.Panic(err)
 				}
 
 				switch status {
-				case issueOpen:
+				case github.IssueOpen:
 					// nothing
-				case issueClosed:
+				case github.IssueClosed:
```

Comment:
super tiny optimization (not important):

``` 
case github.IssueClosed, github.IssueNotFound:
    pass.Reportf(c.Pos(), "invalid TODO: linked issue %s is %s", URL, status)
```

It would apply here and in another similar switch, but I don't think it's super critical.